### DISCUSSION TITLE
Looking for Computational Geometry Developer

### DISCUSSION THREAD

**Discussion by kurt-rhee** [dimforge/nalgebra#1355] [General]
Hey all, 

I am new to this forum, please so please let me know if this kind of post is not welcome here, but my company is looking for a rust developer who is familiar with linear algebra, computational geometry, and has written some algorithms where the time-complexity matters.  We are a startup company that makes simulation software for large solar plants.  (https://plantpredict.com/)

If you are interested in the role, please find me on discord @skrhee or email krhee at terabase.energy!

**Comment by Andlon** [comment]
Hi Kurt, thanks for reaching out. I'd be curious to hear what made you choose Rust and `nalgebra` for your software. Would you be willing to share some of your thoughts on this topic? I'm sure this would be much appreciated among both `nalgebra` developers and users alike.

**Reply by kurt-rhee** [reply]
Hey @Andlon!  Of course, here is some background:

### Why Rust?
Our MVP of the algorithm is written in Python and uses a jit compiler.  This algorithm is too slow and on top of that, the jit compiler only compiles the array-like parts of the code, the sections of code that need variable memory allocation cannot be compiled.  In terms of speed we need a low level language which leaves C++ and Rust as prime candidates.  

Why not C++?  The application we make is a scientific simulation application developed by both domain experts and software developers.  Debugging the application can be difficult, since faults can be related to the software or the implementation of the physics model.  Rust ensures that domain experts (who usually come from python) can contribute to the code base without introducing unsafe memory errors.  Incorrect calculations by the simulation tool must be avoided at all costs since our users trust the tool to make multi-million dollar investment decisions.

### Why `nalgebra`
Our shade simulation needs to be geometrically exact rather than discrete (shadow mapping, ray tracing).  Many of the underlying transformations that need to occur are linear algebra related.  `nalgebra` makes it so that we do not have to write those algorithms ourselves.  I personally also very much like the documentation.  In a tool where communication between software developers and domain experts is important, good documentation goes a long way.
